### PR TITLE
Preliminarily build log modules and filled logs in some open apis includes open, close and sync. 

### PR DESCRIPTION
**Pull request by goodenough227**
i have filled logs in projects just as the general bussiness projects preliminarily but there are some question.
As for the apis like put, i noticed that they have been recorded in val so that is it also nacessary to be recorded in log files？
In additon, the v2 has not could be started up by commands in terminal just like redis. Would it be required in the future?
If there are other problems?Reply your opinions, thank you.

### CODE REVIEW

**Comment by roseduan** [comment]
> i have filled logs in projects just as the general bussiness projects preliminarily but there are some question. As for the apis like put, i noticed that they have been recorded in val so that is it also nacessary to be recorded in log files？ In additon, the v2 has not could be started up by commands in terminal just like redis. Would it be required in the future? If there are other problems?Reply your opinions, thank you.

1. not necessary, we just add logs in some important places, such as before some errors are returned.
2. I guess no, lotusdb is just an key value store engine, the redis features will be implemented in renee project.

File: resource/banner/banner.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+ ██         ░████░   ████████  ██    ██   ▒████▒   █████▒    ██████▒
```
Comment (roseduan) [inline resource/banner/banner.txt:1]: this is not necessary

File: logger.go [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+const (
+	DefaultLevel       = zapcore.InfoLevel
+	DefaultTimeLayout  = time.RFC3339
+	DefaultLogDir      = "./log/"
```
Comment (roseduan) [inline logger.go:17]: The log dir will be the same with DirPath in open options.

File: logger.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+	DefaultLogFileName = "log_info.log"
+)
+
+var logger *zap.Logger
+
+// Option custom setup config
+type Option func(*option)
+
+type option struct {
```
Comment (roseduan) [inline logger.go:26]: logOption is better

File: db.go [hunk 4]
```diff
@@ -33,32 +34,54 @@ type Stat struct {
...
+		WithEnableHighlighting(),
+	)
+	if err != nil {
+		Error("open db failed", zap.Error(err))
+	}
+	defer Sync()
+
 	// check whether all options are valid
 	if err := validateOptions(&options); err != nil {
+		Error("db opens failed", zap.Error(err))
```
Comment (roseduan) [inline db.go:55]: not necessary

File: db.go [hunk 5]
```diff
@@ -33,32 +34,54 @@ type Stat struct {
...
 
 	// create file lock, prevent multiple processes from using the same database directory
 	fileLock := flock.New(filepath.Join(options.DirPath, fileLockName))
 	hold, err := fileLock.TryLock()
 	if err != nil {
+		Error("open db failed", zap.Error(err))
 		return nil, err
 	}
 	if !hold {
+		Error("open db failed", zap.Error(ErrDatabaseIsUsing))
```
Comment (roseduan) [inline db.go:77]: not necesssary

File: db.go [hunk 6]
```diff
@@ -33,32 +34,54 @@ type Stat struct {
...
+		WithInfoLevel(),
+		WithFileRotationPath(logPath),
+		WithDisableConsole(),
+		WithTimeLayout("2006-01-02 15:04:05"),
+		WithEnableHighlighting(),
+	)
+	if err != nil {
+		Error("open db failed", zap.Error(err))
+	}
+	defer Sync()
```
Comment (roseduan) [inline db.go:51]: the sync will be called in `Sync` and `Close` function?

Comment (goodenough227) [reply db.go:51]: yeah, i will fix it

**Comment by goodenough227** [comment]
> 
get it

**Comment by roseduan** [comment]
We will rebuild this part later.

**Pull request closed without merging**